### PR TITLE
Created Course Module

### PR DESCRIPTION
**Pull request by Chimexx**
- Created enitity and dto files
- Created migration file
- Added Controller and service for create function

### CODE REVIEW

**Comment by Chimexx** [comment]
@PhilemonBrain, I think course entity should have a userId, i.e the owner of that course, how to I do that?

File: src/courses/dto/update-course.dto.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  rating: number;
+
+  @ApiPropertyOptional()
+  title: string;
+
+  @ApiPropertyOptional()
+  description: string;
+
+  @ApiPropertyOptional()
+  price: number;
```
Comment (SunmadeBabafemi) [inline src/courses/dto/update-course.dto.ts:21]: price can be decimal too e.g $9. 9..... perhaps either here or in the entity file should it be addressed.... when using decimals for price, error:'invalid syntax' is returned

Comment (Chimexx) [reply src/courses/dto/update-course.dto.ts:21]: Well, I dont think anyone will add kobo to the price of anything these days.

Comment (PhilemonBrain) [reply src/courses/dto/update-course.dto.ts:21]: i think the price will rounded to the base like in cents or in kobo, then converted to dollar or naira on the FE. So i think this is okay

File: src/entities/course.entity.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  @Column()
+  duration: number;
+
+  @Column('text', { array: true, nullable: true, default: [] })
+  topics: string[];
+
+  @Column('text', { array: true, nullable: true, default: [] })
+  students: string[];
+
```
Comment (PhilemonBrain) [inline src/entities/course.entity.ts:37]: i think the only compulsory fields should be title, description, and price

Comment (PhilemonBrain) [reply src/entities/course.entity.ts:37]: what do you think?

File: src/courses/dto/update-course.dto.ts (hunk 1 above)
Comment (SunmadeBabafemi) [reply src/courses/dto/update-course.dto.ts:21]: alright sir..

File: src/entities/course.entity.ts (hunk 2 above)
Comment (Chimexx) [reply src/entities/course.entity.ts:37]: That means it will be fine to have a course without the tutor's name? Language can be nullable

**Review by SunmadeBabafemi — APPROVED** [review]

**Pull request merged**